backupccl: fix backup protected timestamp test

pbardea opened the pull request:
The protected timestamp test for backup only tested that the timestamp
was released after the backup, not if the timestamp was actually
protected in the first place. This commit adds an AOST select at what
should otherwise be a timestamp that has been GC'd.

The test will fail on this SELECT command if the timestamp is not
protected.

Fixes #51068.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51072)
<!-- Reviewable:end -->

dt commented on pkg/ccl/backupccl/backup_test.go:
```diff
@@ -3869,6 +3871,7 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 
 	// We should have refused to GC over the timestamp which we needed to protect.
 	gcTable(true /* skipShouldQueue */)
+	runner.Exec(t, "SELECT * FROM foo AS OF SYSTEM TIME "+tsBefore)
```
huh, is this require to get it to fail if the backup doesn't protect its TS? I'd have expected the ExportRequest to hit it when we close allowResponse below, right? Oh, I wonder if it is because we're using a *Response* filter, not a Request filter, so it has eagerly executed before GC?

pbardea replied:
I ~think that it is required to fail with `kv.gc_ttl.strict_enforcement.enabled` which is set by default. I would prefer to test this more explicitly with an ExportRequest hitting the GC threshold. Will try updating it with a request filter.

pbardea committed:
backupccl: fix protected timestamp test

Previously the protected timestamp test would not fail if the backup
did not protect its timestamp. This change ensures that the backup gets
blocked on the ExportRequest and that the error that is eventually
returned by the BACKUP command fails the test.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51072#-:-MC7NyfS7ZAk2lMFnmmM:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner and @dt)

___
*[pkg/ccl/backupccl/backup_test.go, line 4142 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/51072#-MC7NzLuB8i5qj_wrFl-:-MC7NzLuB8i5qj_wrFl0:b26crwt) ([raw file](https://github.com/cockroachdb/cockroach/blob/918dbe88510cd9f7b25d879cf68bd043040169e9/pkg/ccl/backupccl/backup_test.go#L4142)):*
> ```Go
> 	rowCount := runner.QueryStr(t, "SELECT * FROM foo")
> 
> 	g, _ := errgroup.WithContext(ctx)
> ```

this works but a `chan error` would also have worked. 


<!-- Sent from Reviewable.io -->

pbardea commented:
TFTRs
bors r=ajwerner,dt

pbardea commented:
Looks like bors got canceled. Trying again.
bors r=ajwerner,dt
